api/src/controllers/extensions.ts

```diff
@@ -45,7 +45,7 @@ router.get(
 		}
 
 		res.setHeader('Content-Type', 'application/javascript; charset=UTF-8');
-		res.setHeader('Cache-Control', getCacheControlHeader(req, ms(env.EXTENSIONS_CACHE_TTL as string), false, false));
+		res.setHeader('Cache-Control', getCacheControlHeader(req, getMilliseconds(env.EXTENSIONS_CACHE_TTL), false, false));
```

Comment:
(pardon the unprompted review/comment here, which is sparked by https://github.com/directus/directus/pull/17464/files#r1108236068)

Currently `getCacheControlHeader` will early return `no-cache` when the ttl is null or less than 0: https://github.com/directus/directus/blob/37faf3bd5d0978812362d1803b4e2d6c2ce2d9f6/api/src/utils/get-cache-headers.ts#L24-L25

However if we use `getMilliseconds()` on undefined values such as `EXTENSIONS_CACHE_TTL`, it now defaults to 0. This means the "null or less than 0" early return logic is technically never reached now if I'm not mistaken. 

This endpoint also technically used to return `no-store` explicitly when `EXTENSIONS_CACHE_TTL` was undefined with a ternary operator. 

Will this cause any unintended behaviors somehow? I'm not 100% sure myself, so just wanted to ask about it as additional sanity checks. 🤔

---

Though understandably the app currently always sends `no-store`:

https://github.com/directus/directus/blob/37faf3bd5d0978812362d1803b4e2d6c2ce2d9f6/app/src/api.ts#L8-L14

thus `getCacheControlHeader()` will still return `no-store` with the current changes.

Reply:
> (pardon the unprompted review/comment here, which is sparked by https://github.com/directus/directus/pull/17464/files#r1108236068)

You're review is very much appreciated! Thanks! 😃  

> Currently `getCacheControlHeader` will early return `no-cache` when the ttl is null or less than 0:
> 
> https://github.com/directus/directus/blob/37faf3bd5d0978812362d1803b4e2d6c2ce2d9f6/api/src/utils/get-cache-headers.ts#L24-L25
> 
> However if we use `getMilliseconds()` on undefined values such as `EXTENSIONS_CACHE_TTL`, it now defaults to 0. This means the "null or less than 0" early return logic is technically never reached now if I'm not mistaken.

Good catch! I must have overlooked that.
As `max-age: 0` and `no-cache` doesn't necessarily mean exactly the same (see e.g. https://stackoverflow.com/a/1383359/20504834) I wanted to allow `0` as a valid max-age value in `getCacheControlHeader`.

I wonder what the best solution would be.
- Should we instead return `null` for undefined values in `getMilliseconds()`? Would that still be safe?
- Should we add direct fallbacks everywhere where appropriate? `env.EXTENSIONS_CACHE_TTL ? getMilliseconds(env.EXTENSIONS_CACHE_TTL) : null` (would defeat my secondary goal of simplifying the code with the introducing of `getMilliseconds()`)

> This endpoint also technically used to return `no-store` explicitly when `EXTENSIONS_CACHE_TTL` was undefined with a ternary operator.
> 
> Will this cause any unintended behaviors somehow? I'm not 100% sure myself, so just wanted to ask about it as additional sanity checks. 🤔

I changed this on purpose because I see no reason to use `no-store` here. My tests have shown that changes to extensions with `no-cache` are successfully detected by the browser.

> Though understandably the app currently always sends `no-store`:
> 
> https://github.com/directus/directus/blob/37faf3bd5d0978812362d1803b4e2d6c2ce2d9f6/app/src/api.ts#L8-L14
> 
> thus `getCacheControlHeader()` will still return `no-store` with the current changes.

Hmm, interesting! Is this still appropriate? 🤔

Reply:
> @paescuj There is a difference between `no-store` and `no-cache`.

That's true. But `no-cache` should actually be enough for the extensions endpoint, right? We're using `no-cache` for the API responses as well and my local tests were successful. 

> I think returning `null` or throwing an error on invalid string inputs in the function is appropriate.

I actually wanted to prevent invalid values from throwing any errors and blocking things, so I'll see whether returning `null` can be safely used at those places 👍

Reply:
> That's true. But no-cache should actually be enough for the extensions endpoint, right? We're using no-cache for the API responses as well and my local tests were successful.

Ah yeah for extensions it's enough 👍🏻

Reply:
The `getMilliseconds()` function now fallback to `undefined` by default instead of `0` while the fallback is customizable so it fits in every place.